docs/reference/commandline/pull.md

```diff
@@ -45,7 +45,7 @@ use `docker pull`:
     # and the un-tarred base).
     $ docker pull --all-tags centos
     # will pull all the images from the centos repository
-    $ docker pull registry.hub.docker.com/debian
+    $ docker pull docker.io/library/debian
```

Comment:
IIUC, the example here was to illustrate that a domain can be part of the "pull", I don't think we have other examples illustrating that; what's your suggestion to illustrate that?

For the other references to hub.docker.com; already created  PR for those :smile: https://github.com/docker/docker/pull/20946